warehouse/templates/manage/release.html

```diff
@@ -91,6 +91,7 @@ <h2 class="heading-wsubtitle__heading">Release version {{ release.version }}</h2
                 </a>
               </li>
 
+              {% if not disallow_deletion %}
```

Comment:
This should probably be inverted to match the behavior below for "Delete release":

```
{% if disallow_deletion %}
    <p>....message about release deletion being temporarily disabled....</p>
{% else %}
    normal behavior
{% endif %}

Reply:
I think instead it'd be better to leave the templates unchanged and instead, and just let the request happen and block it in the "delete" view instead (with a flash indicating that it failed), similar to how we do for user registration and read-only mode.

Reply:
Make changes as per @di